src/hotspot/share/runtime/thread.cpp

```diff
@@ -854,7 +854,7 @@ void JavaThread::collect_counters(jlong* array, int length) {
   for (int i = 0; i < length; i++) {
     array[i] = _jvmci_old_thread_counters[i];
   }
-  for (JavaThreadIteratorWithHandle jtiwh; JavaThread *tp = jtiwh.next(); ) {
+  for (JavaThread* tp : ThreadsListHandle()) {
```

Comment:
This is a nicer interface with underlying C++ magic.  There are lots of these in the source code.  Is it the intention to replace these as we go?

Reply:
There's definitely lots of these:
```
$ grep -r JavaThreadIteratorWithHandle src/hotspot | sed 's/:.*//' | sort | uniq -c
   1 src/hotspot/os/linux/os_linux.cpp
   1 src/hotspot/share/compiler/compileBroker.cpp
   1 src/hotspot/share/gc/parallel/mutableNUMASpace.cpp
   2 src/hotspot/share/gc/shared/collectedHeap.cpp
   1 src/hotspot/share/gc/shared/gcLocker.cpp
   1 src/hotspot/share/gc/shenandoah/shenandoahCodeRoots.cpp
   5 src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp
   2 src/hotspot/share/gc/shenandoah/shenandoahPacer.cpp
   1 src/hotspot/share/jfr/utilities/jfrThreadIterator.cpp
   1 src/hotspot/share/jfr/utilities/jfrThreadIterator.hpp
   1 src/hotspot/share/opto/idealGraphPrinter.cpp
   1 src/hotspot/share/prims/jvmtiEnvBase.cpp
   1 src/hotspot/share/prims/jvmtiEventController.cpp
   1 src/hotspot/share/prims/jvmtiImpl.cpp
   1 src/hotspot/share/prims/jvmtiTagMap.cpp
   1 src/hotspot/share/prims/whitebox.cpp
   5 src/hotspot/share/runtime/escapeBarrier.cpp
   2 src/hotspot/share/runtime/handshake.cpp
   1 src/hotspot/share/runtime/os.cpp
   8 src/hotspot/share/runtime/safepoint.cpp
   1 src/hotspot/share/runtime/thread.cpp
   3 src/hotspot/share/runtime/threadSMR.hpp
   3 src/hotspot/share/runtime/vmOperations.cpp
   1 src/hotspot/share/services/heapDumper.cpp
   1 src/hotspot/share/services/management.cpp
   1 src/hotspot/share/services/threadService.cpp
   1 src/hotspot/share/utilities/globalCounter.cpp
```

I believe the plan is to migrate to this newer, cleaner interface.
Obviously not with this bug, but down the road...

Reply:
Awesome, thanks!